packages/web-components/src/components/ic-checkbox/ic-checkbox.tsx

```diff
@@ -127,6 +128,9 @@ export class Checkbox {
 
   componentWillLoad(): void {
     addFormResetListener(this.host, this.handleFormReset);
+    this.host
+      .querySelector("ic-text-field")
+      ?.addEventListener("icChange", (e) => e.stopImmediatePropagation());
```

Comment:
How does a consumer grab the value from the text-field and know it's related to a particular checkbox?

Reply:
I wasn't aware that was previously a feature. This code only prevents `ic-text-field`'s onChange from bubbling up to where it was being caught by the onIcChange of `ic-checkbox-group`. They could still put an onIcChange on the text-field itself and then check if the element's parent is a checkbox or not.

Reply:
I guess that is where the idea for the object below comes into play....
```
[
    { label: 'item-1', value: 'item-1', disabled: true, other: null, checked: false },
    { label: 'item-2', value: 'item-2', disabled: false, other: null, checked: true },
    { label: 'item-3', value: 'item-3', disabled: false, other: null, checked: true },
]
```

Is it possible to add something to the checkedOptions array to include the textfield value?

Reply:
Would be good for the icCheck event to emit the checked value - struggling to get the checked value of a checkbox through the event

Reply:
Have updated the stories with an example. Also added another attribute to show which checkbox was pressed

Reply:
Can see that a `textfieldValues` nd `checkedOptions` array has been added to the `icChange` event and the order of textfield values match. 
Could we merge the two together as they should line up based on the index?